src/main/java/com/example/main/SampleApplication.java

```diff
@@ -47,17 +63,60 @@ public class SampleApplication implements CommandLineRunner {
     @Autowired
     private ApplicationContext applicationContext;
 
+    @Autowired
+    private CosmosTemplate cosmosTemplate;
+
+    @Autowired
+    private ReactiveCosmosTemplate reactiveCosmosTemplate;
+
     public static void main(String[] args) {
         SpringApplication.run(SampleApplication.class, args);
     }
 
     public void run(String... var1) {
 
-        final User1 testUser11 = new User1("testId1", "testFirstName", "testLastName1");
-        final User1 testUser12 = new User1("testId2", "testFirstName", "testLastName2");
+        ChangeFeedProcessorOptions changeFeedProcessorOptions = new ChangeFeedProcessorOptions();
+        changeFeedProcessorOptions.setStartFromBeginning(true);
+        changeFeedProcessorOptions.setLeasePrefix("SOME_PREFIX");
+        changeFeedProcessorOptions.setFeedPollDelay(Duration.ofMillis(0L).plusSeconds(5));
+
+        CosmosAsyncClient cac = primaryDataSourceConfiguration
+                .primaryCosmosClientBuilder(new CosmosProperties()).buildAsyncClient();
+        CosmosAsyncContainer cosmosMonitoredContainer = cac.getDatabase(primaryDataSourceConfiguration.getDatabaseName())
+                .getContainer(reactiveCosmosTemplate.getContainerName(User1.class));
+        cac.getDatabase(primaryDataSourceConfiguration.getDatabaseName())
+                .createContainerIfNotExists("targetContainer", "/id").block();
+        CosmosAsyncContainer cosmosTargetContainer = cac.getDatabase(primaryDataSourceConfiguration.getDatabaseName())
+                .getContainer("targetContainer");
+        cac.getDatabase(primaryDataSourceConfiguration.getDatabaseName())
+                .createContainerIfNotExists("leaseContainer", "/id").block();
+        CosmosAsyncContainer leaseTargetContainer = cac.getDatabase(primaryDataSourceConfiguration.getDatabaseName())
+                .getContainer("leaseContainer");
+        try {
+            ChangeFeedProcessor cfp = new ChangeFeedProcessorBuilder().options(changeFeedProcessorOptions)
+                .hostName(InetAddress.getLocalHost().getHostName())
+                .feedContainer(cosmosMonitoredContainer).leaseContainer(leaseTargetContainer)
+                .handleChanges((List<JsonNode> docs) -> {
```

Comment:
I don't see us calling the `updateKey()` API here? Isn't customer doing that in the handleChanges() method?

Reply:
No they aren't, they do it by hitting an endpoint:
![image](https://github.com/kushagraThapar/azure-spring-data-cosmos-demo/assets/63077034/d55b7b5c-1ecb-4517-826e-c347b60e1939)

Reply:
Got it, thanks!